llvm/lib/TargetParser/TargetParser.cpp

```diff
@@ -253,6 +274,12 @@ AMDGPU::IsaVersion AMDGPU::getIsaVersion(StringRef GPU) {
   case GK_GFX1151: return {11, 5, 1};
   case GK_GFX1200: return {12, 0, 0};
   case GK_GFX1201: return {12, 0, 1};
+
+  // Generic targets use the earliest ISA version in their group.
```

Comment:
Should it use something else instead of the earliest ISA version? I am not sure what it would be, but it feels uneasy for some reason.

If we leave it "as is", can this be misused? E.g. getIsaVersion will return {9,0,0} for gfx9-generic, which got the madmix instructions, which is not in {9,0,4}.

Reply:
- maybe UINT32_MAX for the minor and stepping for gfx9-generic, gfx11-generic
- maybe UINT32_MAX for the stepping for gfx10.1 and gfx10.3

thoughts?

Reply:
I think it's alright as is, but this API is bad and should probably be refactored IMO. Most users of the API are just interested in checking the version major, sometimes minor (10.1 vs 10.3).

In theory, this API should _never_ be used to check for presence of a feature, that's always done through the feature list check, so it shouldn't really be abusable. I added a comment though to revisit this and make the intent clearer.

Reply:
Ok, works for me for now.

> should probably be refactored IMO

Agreed.